src/dbnode/storage/shard.go

```diff
@@ -499,7 +498,7 @@ func (s *dbShard) OnRetrieveBlock(
 		return
 	}
 
-	entry, err = s.newShardEntry(id, newTagsIterArg(tags))
```

Comment:
Can we now drop `newTagsIterArg` / `newTagsArg` / `tagsArgOptions` completely?

Reply:
I think we can!